common/djangoapps/edxmako/services.py

```diff
@@ -0,0 +1,36 @@
+"""
+Supports rendering an XBlock to HTML using mako templates.
+"""
+
+import warnings
+from xblock.reference.plugins import Service
+
+
+class MakoService(Service):
+    """
+    A service for rendering XBlocks to HTML using mako templates.
+
+    Args:
+        render_template(function): function that renders the given context to the given template.
+    """
+    def __init__(
+        self,
+        render_template,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self._render_template = render_template
+
+    @property
+    def render_template(self):
+        """
+        Returns the XBlock render_template function.
+
+        Function takes (template, context) and returns rendered HTML.
+        """
+        warnings.warn(
+            'Use of MakoService.render_template is deprecated. '
+            'Use xblockutils.resources.ResourceLoader.render_mako_template or a JavaScript-based template instead.',
+            DeprecationWarning, stacklevel=2,
+        )
+        return self._render_template
```

Comment:
Instead of passing in the `render_template` method, can we instead have a parameter `namespace_prefix=''`? And then instead of a property this can become:

```
def render_template(template_name, dictionary, namespace='main')
  return render_to_string(template_name, dictionary, namespace=self.namespace_prefix + namespace)
```

That way we don't need to pass in the whole code to the service but just the relevant parameter.

Reply:
Sure thing: https://github.com/edx/edx-platform/pull/28571/commits/ae32582477ca80905e4555e3f283027ff40351c5#diff-0f9960747639224845a14d42acee7d24eae92d50406e88a3142a7da45467c596R215